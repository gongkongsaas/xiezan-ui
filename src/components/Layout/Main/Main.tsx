import { defineComponent } from 'vue'

export default defineComponent({
  setup(props, { slots }) {
    return () => (
      <div class="main-box">
        <main class="main">
          <div class="main-inner">{slots.default && slots.default()}</div>
        </main>
      </div>
    )
  },
})
