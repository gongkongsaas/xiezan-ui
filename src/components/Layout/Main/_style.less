.main-box {
  display: flex;
  flex-grow: 1;
  padding-left: 8px;
  background-color: #f5f5f5;
  .main {
    width: 100%;
    background: #fff;
    overflow: auto;
    .main-inner {
      min-height: 100%;
      padding: 0 16px;
      border-right: 8px #f5f5f5 solid;
      overflow: hidden;
    }
  }  
}
