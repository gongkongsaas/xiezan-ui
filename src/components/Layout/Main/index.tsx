import Footer from './main'

export default Footer
