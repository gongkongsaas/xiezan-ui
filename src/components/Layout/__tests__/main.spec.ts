import { shallowMount } from '@vue/test-utils'
import Main from '../main'

describe('Main', () => {
  it('slot is health', () => {
    const wrapper = shallowMount(Main, {
      slots: {
        default: () => 'testSlot',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
