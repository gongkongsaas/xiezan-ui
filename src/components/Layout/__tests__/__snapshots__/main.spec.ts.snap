// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main slot is health 1`] = `
<div class="main-box">
  <main class="main">
    <div class="main-inner">testSlot</div>
  </main>
</div>
`;
